src/vaadin-details.html

```diff
@@ -143,6 +143,24 @@
           return this.shadowRoot.querySelector('[part="summary"]');
         }
 
+        /**
+         * Override logic from vaadin-control-state-mixin to only focus the summary element
+         * when using keyboard navigation. This avoids issues when clicking into the details,
+         * such as text selection and click events not working
+         * @override
+         * @protected
+         */
+        _focus() {
+          if (!this.focusElement || this._isShiftTabbing) {
+            return;
+          }
+
+          if (this._keyboardActive) {
+            this.focusElement.focus();
+          }
+          this._setFocused(true);
+        }
```

Comment:
This just duplicates the logic from `ControlStateMixin`, while adding the check for `this._keyboardActive` on line `158`.